sphinx_needs/filter_common.py

```diff
@@ -297,8 +298,65 @@ def intersection_of_need_results(list_a: list[T], list_b: list[T]) -> list[T]:
     return [a for a in list_a if a in list_b]
 
 
+def filter_needs_mutable(
+    needs: NeedsMutable,
+    config: NeedsSphinxConfig,
+    filter_string: None | str = "",
+    current_need: NeedsInfoType | None = None,
+    *,
+    location: tuple[str, int | None] | nodes.Node | None = None,
+    append_warning: str = "",
+) -> list[NeedsInfoType]:
+    return _filter_needs(
+        needs.values(),
+        config,
+        filter_string,
+        current_need,
+        location=location,
+        append_warning=append_warning,
+    )
+
+
+def filter_needs_view(
+    needs: NeedsView,
+    config: NeedsSphinxConfig,
+    filter_string: None | str = "",
+    current_need: NeedsInfoType | None = None,
+    *,
+    location: tuple[str, int | None] | nodes.Node | None = None,
+    append_warning: str = "",
+) -> list[NeedsInfoType]:
+    return _filter_needs(
+        needs.values(),
+        config,
+        filter_string,
+        current_need,
+        location=location,
+        append_warning=append_warning,
+    )
+
+
+def filter_needs_parts(
+    needs: NeedsPartsView,
+    config: NeedsSphinxConfig,
+    filter_string: None | str = "",
+    current_need: NeedsInfoType | None = None,
+    *,
+    location: tuple[str, int | None] | nodes.Node | None = None,
+    append_warning: str = "",
+) -> list[NeedsInfoType]:
+    return _filter_needs(
+        needs,
+        config,
+        filter_string,
+        current_need,
+        location=location,
+        append_warning=append_warning,
+    )
+
+
 @measure_time("filtering")
-def filter_needs(
+def _filter_needs(
```

Comment:
need to deprecate the old name?

Reply:
I think no, because this is not part of the public api (i.e. https://sphinx-needs.readthedocs.io/en/latest/api.html), and anyhow it will force people to stop using it if they were 😅

Reply:
(in principle we should change almost every module to have an `_`, but eurgh that would be annoying 😅